### PR TITLE
Larval Hugger rework

### PR DESCRIPTION
**Pull request by destrucktoid**
## About The Pull Request
Completely reworks how larval huggers work, removing chestbursting entirely and replacing it with passive larva and psy point generation. At later embryo levels marines will start to ocasionally take damage and fall unconscious.

## Why It's Good For The Game

I spoke to a lot of people about my earlier PR for huggers and was told that Larval huggers were "basically leftover from capture times". I spoke to people and people seemed to agree that replacing chest bursters with a new passive system would be for the better for both marines and xenomorphs.

Benefits for marines:
You no longer need to instantly medevac if you get facehugged as chestbursters are not a thing.

Benefits for Xenomorphs:
Reliable and consistent larva generation if you are able to get marines with a larva in them to later stages.
Carriers should have more targets for their call of the younger as marines are more likely to stay on the field for longer.
## Changelog
:cl:
add: New passive psy/embryo point generation
del: No more chestbursters
/:cl:

### CODE REVIEW

**Comment by Kuro020** [comment]
Can you post how many points it generates?

**Comment by destrucktoid** [comment]
> Can you post how many points it generates?

6 stages of larva:
1: 0.0003
2: 0.0006
3: 0.0012
4: 0.0024
5: 0.0048
6: 0.0096

Per marine, you should get 1 larva after about 15-20 minutes.

**Comment by iforgotmeword** [comment]
make it so marines that are infected also infections over stage 3 and get clone loss over 5
so there is a still a personal reason to get med-evac 
that or make call of the younger WAYYYY stronger (like a tile pull of 3  +1 per stage )

because lets be real if marines didn't have a "YOUR GOING TO DIE IF YOU DON'T GET THIS FIXED" 
they aren't going up no matter how much profit xeno will make from them (see organ damage and 60 cloneloss still not going up)

also rip capture gameplay :(
Capturegameplay

**Comment by iforgotmeword** [comment]
also personal thing uh add a note why the flare code has a check so no one goes WIF? when looking at the code

**Comment by destrucktoid** [comment]
> because lets be real if marines didn't have a "YOUR GOING TO DIE IF YOU DON'T GET THIS FIXED" they aren't going up no matter how much profit xeno will make from them (see organ damage and 60 cloneloss still not going up)

I've made it so when you are at stage 6 you can fall unconscious for 4 seconds with a 0.5% chance of happening every 2 seconds

**Comment by destrucktoid** [comment]
> > Can you post how many points it generates?
> 
> 6 stages of Embryo: 1: 0.0003 2: 0.0006 3: 0.0012 4: 0.0024 5: 0.0048 6: 0.0096
> 
> Per marine, you should get 1 larva after about 15-20 minutes.

HAHAHA NOT ANYMORE BECAUSE PROCESSING IS EVERY 2 SECS NOT EVERY SECOND SO I HAD TO DOUBLE IT.
Sometimes i hate coding.
Basically just double all those numbers i gave you before. Should still be the same as before though in terms of return on investment

**Comment by XElectricX** [comment]
This kills the surgeon.

**Comment by destrucktoid** [comment]
> This kills the surgeon.
 
No because marines _should_ still want to go get the hugger removed in order to deny xenos resources. Are they likely to do so? Probably not. Either way though a surgeon would be required to remove them.

**Comment by Naaanii** [comment]
> > This kills the surgeon.
> 
> No because marines _should_ still want to go get the hugger removed in order to deny xenos resources. Are they likely to do so? Probably not. Either way though a surgeon would be required to remove them.

Not going up or getting surgery vs a larva every 15-20 minutes is a deal marines will take any time.

This removes the whole point of larval huggers other than the stun and completely nukes the possibility of a shipside drone, while which is rare, is still possible.

A similar PR was also denied <https://github.com/tgstation/TerraGov-Marine-Corps/pull/14245>

**Comment by destrucktoid** [comment]
> > > This kills the surgeon.
> > 
> > 
> > No because marines _should_ still want to go get the hugger removed in order to deny xenos resources. Are they likely to do so? Probably not. Either way though a surgeon would be required to remove them.
> 
> Not going up or getting surgery vs a larva every 15-20 minutes is a deal marines will take any time.
> 
> This removes the whole point of larval huggers other than the stun and completely nukes the possibility of a shipside drone, while which is rare, is still possible.
> 
> A similar PR was also denied #14245

That PR is not similar in the slightest. It gave a flat reward of 25 psy points for getting a impregnation off. This PR punishes marines for staying groundside and not getting rid of the larva by making xenos gain more larva and psy points. You say it removes the whole point of larva huggers but it doesn't. _A good marine after this change will **still** go to the ship and get the larva removed at some point_. More importantly it actually makes carrier better in combat if a marine fails to go get the embryo removed as in doing so call of the hugger will be usable on them. This means that you actually now have more utility in combat.

**Comment by Raven-Industries** [comment]
Balance concerns aside, it does seem a shame to lose this connection to the source material. TGMC is hardly a purist in that regard, but chest-bursting is such an iconic part of the larval hugger; my own sense (and I don't know if others would share it) is that it would make the game slightly, qualitatively, lesser.

**Comment by Coastfront** [comment]
And at last; Petals reminding the bird of a life as a seed rot away for feathers.
The final green parts for what is now; To die with memories of what was thus.

I personally, do not like abandoning possibly the last actual flavor for classic aliens that we have, aside from the queen, drone, & title of 'xenomorphs."

And otherwise; I do infact agree, that most marines would literally never go shipside unless if you point the threat of instant, inevitably death at them, by virtue of the amount of marines I see 100% ignore 3 fractures & IB to continue ungaing because "I'm not dead yet."

**Comment by DrAmazing343** [comment]
> Balance concerns aside, it does seem a shame to lose this connection to the source material. TGMC is hardly a purist in that regard, but chest-bursting is such an iconic part of the larval hugger; my own sense (and I don't know if others would share it) is that it would make the game slightly, qualitatively, lesser.

I agree with this sentiment wholly. I understand the purpose, gameplay-wise, of the Carrier's larval hugger in modern-day TGMC, but at the same time I can hardly sit down and ignore the fact that this is one of the last true "Alien" things we have left. Is it exaggerated for gameplay? Of course! The whole game is. We don't play a round out over twelve to twenty-four hours, so we can't realistically simulate the hugger and the horror of the implanting, but we CAN keep around the gameplay mechanic that at least reaches for part of that experience.

It would be a shame to see that go in favor of gameplay, as much as I nonetheless understand and appreciate the improvement to Carrier viability this could bring.

**Comment by Kuro020** [comment]
Balance and gameplay is more important than flavour or trying to retain ties to Aliens as a series (which we dont, the power rangers we have for marines running should tell you so), so i'm not gonna touch on that point much.

Although the pr is disliked as seen by downvotes, the point the author is making is valid, chestburster is a holdover from capture gameplay era, currently its only useful for the stun and stamina drain, chestbursting itself is rare and making it generate points is more inline with the xeno gameplay we have today.

Marines not going shipside (we ignoring tad and apc surgery here?) because this doesnt kill them is a skill issue and i will not consider marine player's room temperature IQ when making balance decisions.

Overall im tending towards approving this, i have not seen any good arguments against this pr yet. I'll give it 1 more day to see if anyone has any actual reasons on why this is bad.

**Comment by XElectricX** [comment]
> Marines not going shipside (we ignoring tad and apc surgery here?)

Shipside isn't what I care about. Even groundside people just want a splint slapped on  and go "I'm fine" when they have internal bleeding or organ damage. 

As I mentioned before, rest of the PR is ok. Just leave chest bursting in, it doesn't need to give points or anything. It's a neat mechanic that puts a marine on a death timer (not as punishing either, doesn't even perma anymore and you can't inject larval growth!). Not to mention you can avoid being impregnated by just being set on fire these days.

**Comment by Snickeydoo** [comment]
Random nobody here that’s played on and off.

Why not remove chestbursting as it is now but bring it back in a different way? I can see there have been many more changes even compared to when I last played and as others have said I can also see that being an aliens vs marines immersive sim is not the primary goal.

But as a happy medium (And I am just throwing shit at the wall here) perhaps carrier could have something added to it that allows it to MAKE infection lethal if infected marines stick around the carrier for too long, like starting the detonation timer of a bomb (Burst timer itself would start alot lower than it is ofc) that ticks down faster the closer the victim is.

I feel like that saw trap esque pressure can be translated into a fun gameplay scenario where its less about the long horror and more like infected marines are given an oh shit moment and are forced to make a decision or something. Again it doesn’t have to be carrier and maybe the “bomb” in question is more literal and they ACTUALLY explode (like facehuggers implanting “natural” explosives into their hosts)

I’m a nobody but I felt I had an opinion to share here.

**Comment by destrucktoid** [comment]
> perhaps carrier could have something added to it that allows it to MAKE infection lethal if infected marines stick around the carrier for too long

That's literally call of the younger. It's such a slept on ability i reckon next to nobody remembers it exists. If you are a marine and you go near a carrier whilst pregnant they can just pull you 3 tiles towards them and stun you for like 1-2 seconds as well as giving embryo progression. No it's not a "You will instantly die" thing but it IS a big threat.

**Comment by Snickeydoo** [comment]
I did read that on the wiki yeah, But I was thinking more of a bomb that acts as a deterrent and if the deterrent warning is ignored… Organic bomb from within. And maybe there can be on the field methods to neutralize said bomb once it’s active instead of running back up to the ship and removing it entirely?

I’m just putting in ideas for it but realistically I think a lot of players just really like that morbid tension from the franchise of having an egg inside that threatens them, I feel like its kind of what makes REGULAR facehuggers scary and is part of their unique identity compared to Xeno’s.

I like the idea of being infected regularly just passively benefiting the other side and eventually still making them sick if its taken too far, But I also think a happy medium can be met at some point and in some way so that  the regular facehugger’s literal base identity and what they’re known for can stay somewhere in the game even if its only attached to a specific caste and removed from the Xeno lifecycle.

As you can see I am quite the yapper, FYI you can tell me to shut up at literally any point and I will.

**Review by Kuro020 — APPROVED** [review]
good to go for a TM

File: code/modules/mob/living/carbon/xenomorph/embryo.dm [hunk 1]
```diff
@@ -16,6 +16,7 @@
 	var/boost_timer = 0
 	var/hivenumber = XENO_HIVE_NORMAL
 	var/admin = FALSE
+	var/datum/job/xeno_job
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/embryo.dm:19]: autodoc, although I don't see why this is even a var at all.

**Review by Lumipharon — CHANGES_REQUESTED** [review]
This change is pretty smelly tbh.
Bursts no longer perma a marine, but they are still important because they force marines out of the fight temporarily for surgery. This is a specific form of attrition, and is far more valuable that fractional points gen, especially when xenos so frequently have issues of excess burrowed.

**Comment by Ldip999** [comment]
They do perma a marine, you might be reviveable, but your soul can no longer reenter your body. PRobably unintended and just a bug, but they do perma none the less.
Source: Happened to me a couple times.

**Comment by Naaanii** [comment]
> They do perma a marine, you might be reviveable, but your soul can no longer reenter your body. PRobably unintended and just a bug, but they do perma none the less. Source: Happened to me a couple times.

They perma you if you take control of the larva and have Xeno preferences on. Otherwise you should be revivable and free to reenter your body.

**Comment by Ldip999** [comment]
Either way if it's meant to give me the larva or if it's meant to make me reviveable, it instead picked someone from the larva queue and it made me reviveable, leading me to get round removed. This behaviour is probably not intended, but it is the reality of the current state of affairs.

**Review by TiviPlus — APPROVED** [review]
Bad change, silent sabotaging your team for bald player is stupid especially since bursts are revivable

Coke ok

**Review by Lumipharon — APPROVED** [review]

**Comment by Naaanii** [comment]
> Bad change, silent sabotaging your team for bald player is stupid especially since bursts are revivable
> 
> Coke ok

image

**Comment by Technical-l** [comment]
> > I personally, do not like abandoning possibly the last actual flavor for classic aliens that we have, aside from the queen, drone, & title of 'xenomorphs."
> 
> I'd like to point out that canonically a chestburster takes 8-12 hours (can vary above and below) to incubate, sometimes longer. In the timeframes we see in game it's unrealistic for anyone to ever actually burst.

aint no way you bringing out the realism argument here. Jesus christ bro, imagine waiting even a single for someone to burst, it'd absolutely nerf huggers into the ground. You can't bring up the realism argument here, especially considering the game as a whole takes liberties to improve gameplay at the cost of realism. Besides, staying at least somewhat true to source material and having a """""realistic"""" game are two completely different things.

**Comment by zzurge** [comment]
Just because you can,doesnt mean you should

File: code/modules/mob/living/carbon/xenomorph/embryo.dm [hunk 2]
```diff
@@ -16,7 +16,7 @@
 	var/boost_timer = 0
 	var/hivenumber = XENO_HIVE_NORMAL
 	var/admin = FALSE
-
+	var/mob/living/carbon/xenomorph/source
```
Comment (Lumipharon) [inline code/modules/mob/living/carbon/xenomorph/embryo.dm:19]: autodoc.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/embryo.dm (hunk 2 above)
Comment (Ldip999) [reply code/modules/mob/living/carbon/xenomorph/embryo.dm:20]: Sorry

**Review by Lumipharon — APPROVED** [review]

**Comment by Wisemonster** [comment]
You should add that the embryo will cause knockout at later stages in the pr's description/changelog, people currently think that this is completely ignorable at all stages cause "There's no noticeable effect"

**Comment by destrucktoid** [comment]
> You should add that the embryo will cause knockout at later stages in the pr's description/changelog, people currently think that this is completely ignorable at all stages cause "There's no noticeable effect"

Sure

**Comment by destrucktoid** [comment]
_Sigh_, I just changed the description a bit and TGS candidate gets removed... Need to wait for a maint to give it back then

**Comment by HenryOfSkalitz** [comment]
Five hundred larval points

**Comment by Ldip999** [comment]
Me when skibidi squad doesn't go shipside for surgery even though I yell at them to (They are paid xeno agents generating 500 larba points a second)

**Comment by Arturlang** [comment]
What if they don't pop groundside, but they do shipside?

**Comment by Ldip999** [comment]
> What if they don't pop groundside, but they do shipside?



> This kills the surgeon.

**Comment by RipGrayson** [comment]
I don't like this, bursting is a cool mechanic and being infected with a larval hugger should be immediate shipside evac, not a brief interruption followed by a mild condition that marines can address whenever they feel like they have the time.

**Comment by Looking4bros** [comment]
Without chestbursting, i never would have been able to infect myself with the larval hugger i got from the Christmas tree and kill the round by spliting all the xeno statics into greenos.

**Pull request closed without merging**